sdk/keyvault/keyvault-admin/test/utils/common.ts

```diff
@@ -56,3 +60,33 @@ export function getSasToken() {
 
   return { blobStorageUri, blobSasToken };
 }
+
+/**
+ * The known API versions that we support.
+ */
+export const serviceVersions = ["7.0", "7.1", "7.2", "7.3-preview"] as const;
+
+/**
+ * Fetches the service version to test against. This version could be configured as part of CI
+ * and then passed through the environment in order to support testing prior service versions.
+ * @returns - The service version to test
+ */
+export function getServiceVersion(): NonNullable<
+  (AccessControlClientOptions | KeyVaultBackupClientOptions)["serviceVersion"]
```

Comment:
Both of them extend `CommonClientOptions`, but `CommonClientOptions` does not have the `serviceVersion` property. I could move `serviceVersion` to the `CommonClientOptions` ,but that would be a breaking change, I think 🤔

Reply:
Because the type `SUPPORTED_API_VERSIONS` is already available here, we probably don't even need to infer from the serviceVersion parameter. Something like this should work:

```ts

 * Fetches the service version to test against. This version could be configured as part of CI
 * and then passed through the environment in order to support testing prior service versions.
 * @returns - The service version to test
 */
export function getServiceVersion(): SUPPORTED_API_VERSIONS {
  return env.SERVICE_VERSION || LATEST_API_VERSION;
}

/**
 * A convenience wrapper allowing us to limit service versions without using the `versionsToTest` wrapper.
 *
 * @param supportedVersions - The {@link SupportedVersions} to limit this test against.
 * @param serviceVersion - The service version we want to test support for. If omitted we will default to the version returned from {@link getServiceVersion}.
 * @returns A Mocha Wrapper which will skip or execute the chained tests depending the currently tested service version and the supported versions.
 */
export function onVersions(
  supportedVersions: SupportedVersions,
  serviceVersion?: SUPPORTED_API_VERSIONS[number]
): TestFunctionWrapper {
  return supports(serviceVersion || getServiceVersion(), supportedVersions, serviceVersions);
}

```

Reply:
I like that! thanks!

Reply:
How does this look? https://github.com/Azure/azure-sdk-for-js/pull/19956/commits/3f591b9ee4b53d779be6118ab41eb16c4419c13a